src/client/line.cpp

```diff
@@ -0,0 +1,556 @@
+				if (!params->last_changed_props[prop])
+					continue;
+
+				switch ((LineParams::LineProperty)prop) {
+					case LineParams::LP_TYPE:
+						line_params.type = params->type;
+						break;
+					case LineParams::LP_START_POS:
+						line_params.pos.first = params->pos.first;
+						break;
+					case LineParams::LP_END_POS:
+						line_params.pos.second = params->pos.second;
+						break;
+					case LineParams::LP_START_COLOR:
+						line_params.color.first = params->color.first;
+						break;
+					case LineParams::LP_END_COLOR:
+						line_params.color.second = params->color.second;
+						break;
+					case LineParams::LP_START_ATTACH_TO:
+						line_params.attached_ids.first = params->attached_ids.first;
+						break;
+					case LineParams::LP_END_ATTACH_TO:
+						line_params.attached_ids.second = params->attached_ids.second;
+						break;
+					case LineParams::LP_WIDTH:
+						line_params.width = params->width;
+						break;
+					case LineParams::LP_TEXTURE:
+						line_params.texture = params->texture;
+						m_line_scenenode_list[id]->updateMaterial();
+						break;
+					case LineParams::LP_CLAMP_MODE:
+						line_params.clamp_mode = params->clamp_mode;
+						m_line_scenenode_list[id]->updateMaterial();
+						break;
+					case LineParams::LP_PLAYERNAME:
+						line_params.playername = params->playername;
+						break;
+					case LineParams::LP_ANIMATION:
+						line_params.animation = params->animation;
+						break;
+					case LineParams::LP_BACKFACE_CULLING:
+						line_params.backface_culling = params->backface_culling;
+						m_line_scenenode_list[id]->updateMaterial();
+						break;
+					case LineParams::LP_LIGHT_LEVEL:
+						line_params.light_level = params->light_level;
+						break;
+					case LineParams::LP_AXIS_ANGLE:
+						line_params.axis_angle = params->axis_angle;
+						break;
+					default:
+						break;
+				}
+			}
+			break;
+		}
+		case CE_REMOVE_3DLINE:
+		{
+			remove3DLineFromList(id);
+			break;
+		}
+		default:
+			break;
+	}
+
+	delete params;
+}
+
+void LineSceneNodeManager::step(f32 dtime)
+{
+	static thread_local const float radius =
+			g_settings->getS16("max_block_send_distance") * MAP_BLOCKSIZE * BS;
+
+	v3f ppos = m_env->getLocalPlayer()->getPosition();
+
+	for (auto line : m_line_scenenode_list) {
+		auto &pos = line.second->pos;
+
+		v3f cam_offset = intToFloat(m_env->getCameraOffset(), BS);
+		bool is_far_from_start_pos = ppos.getDistanceFromSQ(pos.first+cam_offset) > radius * radius;
+		bool is_far_from_end_pos = ppos.getDistanceFromSQ(pos.second+cam_offset) > radius * radius;
+
+		bool attachments_invalid = false;
+
+		u16 first_attach_id = line.second->params.attached_ids.first;
+		u16 second_attach_id = line.second->params.attached_ids.second;
+
+		if (first_attach_id && not m_env->getGenericCAO(first_attach_id))
+			attachments_invalid = true;
+
+		if (second_attach_id && not m_env->getGenericCAO(second_attach_id))
+			attachments_invalid = true;
```

Comment:
Please do not use alternative tokens. See [Code style guidelines#Miscellaneous](https://dev.minetest.net/Code_style_guidelines#Miscellaneous) for reference.

```suggestion
		if (first_attach_id && !m_env->getGenericCAO(first_attach_id))
			attachments_invalid = true;

		if (second_attach_id && !m_env->getGenericCAO(second_attach_id))
			attachments_invalid = true;
```